ingester/shell.go

```diff
@@ -258,7 +258,13 @@ func (ingest *Ingester) shell() {
 		// teardown
 		shell.Close()
 	})
-	if err := ssh.ListenAndServe(":2222", nil, ssh.HostKeyFile("sshkey")); err != nil {
-		logrus.Error(err)
+	if cfg.IngesterShell.PasswordLogin {
+		if err := ssh.ListenAndServe("localhost:2222", nil, ssh.HostKeyFile("sshkey"), ssh.PasswordAuth(func(ctx ssh.Context, pass string) bool { return pass == cfg.IngesterShell.Password })); err != nil {
```

Comment:
The SSH host should be configurable as well

Reply:
Do these behaviors look ok:
1) Default for PasswordLogin = False
2) If PasswordLogin == False (or not set), but Password is set, change PasswordLogin = True
3) If PasswordLogin == True, but Password is not set, exit with Error message "Missing Configuration: IngesterShell.Password"
4) Default for SshHostKey = "sshkey"
5) If SshHostKey configured is not a private key file or if it is not present, do not start shell (exit with error message)
Let me know if I need to change any of these.

Reply:
Allow case 2, otherwise those look good